### PR TITLE
Began work on setting up the admin permissions using Devise Authoriza…

### PR DESCRIPTION
**Pull request by abigail29**
Began work on setting up the admin permissions using Devise Authorization!  Thank you!

### CODE REVIEW

File: app/controllers/venues_controller.rb [hunk 1]
```diff
@@ -1,9 +1,19 @@
...
   def show
     @venue = Venue.find(params[:id])
   end
+
+  protected
+  def authorize_user
+    if !user_signed_in? || !current_user.admin?
+      raise ActionController::RoutingError.new("Not Found")
+    end
+  end
```
Comment (bkincart) [inline app/controllers/venues_controller.rb:17]: FYI the way you have this set up, only an admin will be able to see the venues index or show page. Is that what you were going for?

File: db/seeds.rb [hunk 2]
```diff
@@ -5,22 +5,12 @@
...
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
-
-
-Venue.create!( name: 'House of Blues', address: '15 Lansdown Street', description: 'Rock- & blues-themed chain with Southern dishes such as po boys & jambalaya, plus live music.', capacity: 200, url: 'http://www.houseofblues.com/', image: 'https://d2uqqhmijd5j2z.cloudfront.net/files/131579/venue_gallery/HouseOfBluesBostonMA.jpg?1482355562' )
-
-Venue.create!( name: 'Hatch Shell', address: '47 David G Mugar Way', description: 'The Edward A. Hatch Memorial Shell is an outdoor concert venue on the Charles River Esplanade in the Back Bay section of Boston, Massachusetts.', capacity: 500, url: 'http://www.hatchshell.com/', image: 'https://9968c6ef49dc043599a5-e151928c3d69a5a4a2d07a8bf3efa90a.ssl.cf2.rackcdn.com/68958-1.jpg' )
-
+Venue.create!( name: 'House of Blues', address: '15 Lansdown Street', description: 'house o blues yo', capacity: 200, url: 'http://www.houseofblues.com/' )
+Venue.create!( name: 'Hatch Shell', address: '47 David G Mugar Way', description: 'The HALF shell', capacity: 500, url: 'http://www.hatchshell.com/' )
```
Comment (bkincart) [inline db/seeds.rb]: Please indent!

File: app/models/user.rb [hunk 3]
```diff
@@ -3,4 +3,8 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+
+    def admin?
+      role == "admin"
+    end
```
Comment (bkincart) [inline app/models/user.rb]: You should add some validations for the role here: probably a presence validation, plus an inclusion validation to make sure it's one of the valid role types

Comment (bkincart) [reply app/models/user.rb]: You should also add tests for this!

**Review by bkincart — CHANGES_REQUESTED** [review]

**Comment by abigail29** [comment]
This is the updated admin permissions one.  Let me know if anything else is wrong! thank you!

**Review by bkincart — APPROVED** [review]

**Pull request merged**